src/drivers/intel/dmic/dmic.c

```diff
@@ -59,19 +52,29 @@ static const uint32_t base[4] = {PDM0, PDM1, PDM2, PDM3};
 static SHARED_DATA struct dmic_global_shared dmic_global;
 
 /* this ramps volume changes over time */
-static enum task_state dmic_work(void *data)
+static void dmic_gain_ramp(struct dai *dai)
 {
-	struct dai *dai = (struct dai *)data;
 	struct dmic_pdata *dmic = dai_get_drvdata(dai);
 	k_spinlock_key_t key;
 	int32_t gval;
 	uint32_t val;
 	int i;
 
-	dai_dbg(dai, "dmic_work()");
+	/* Currently there's no DMIC HW internal mutings and wait times
+	 * applied into this start sequence. It can be implemented here if
+	 * start of audio capture would contain clicks and/or noise and it
+	 * is not suppressed by gain ramp somewhere in the capture pipe.
+	 */
+
+	dai_dbg(dai, "dmic_gain_ramp()");
 
 	key = k_spin_lock(&dai->lock);
 
+	if (dmic->gain == DMIC_HW_FIR_GAIN_MAX << 11) {
+		k_spin_unlock(&dai->lock, key);
+		return;
+	}
```

Comment:
Could this check and unlock be before spinlock, then just return is enough?

Reply:
The same DAI cannot run on multiple cores, so, I think - yes, we can rely on the fact, that as long as the pipeline task is running, it is safe to access the `dmic` object for reading